modules/programs/radicle.nix

```diff
@@ -0,0 +1,272 @@
+                be detected automatically. The value of this option should likely
+                be the same as the output of
+                `xdg-mime query default x-scheme-handler/https`.
+              '';
+              type = nullOr str;
+              default = detected;
+              defaultText = "Automatically detected browser.";
+              example = "brave.desktop";
+            };
+          };
+      };
+
+      settings = mkOption {
+        default = { };
+        description = "Radicle configuration, written to `~/.radicle/config.json.";
+
+        type = submodule {
+          freeformType = configFile.format.type;
+
+          options.publicExplorer = mkOption {
+            type = str;
+            description = "HTTPS URL pattern used to generate links to view content on Radicle via the browser.";
+            default = "https://app.radicle.xyz/nodes/$host/$rid$path";
+            example = "https://radicle.example.com/nodes/seed.example.com/$rid$path";
+          };
+
+          options.node.alias = mkOption {
+            type = str;
+            description = "Human readable alias for your node.";
+            default = config.home.username;
+            defaultText = lib.literalExpression "config.home.username";
+          };
+          options.node.listen = mkOption {
+            type = listOf str;
+            description = "Addresses to bind to and listen for inbound connections.";
+            default = [ ];
+            example = [ "127.0.0.1:58776" ];
+          };
+        };
+
+      };
+
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      {
+        assertion =
+          !pkgs.hostPlatform.isLinux
+          -> (
+            (filter id [
+              cfg.uri.rad.browser.enable
+              cfg.uri.rad.vscode.enable
+            ]) == [ ]
+          );
+        message = "`rad:`-URI handlers are only supported on Linux.";
+      }
+      {
+        assertion = cfg.uri.web-rad.enable -> cfg.uri.web-rad.browser != null;
+        message = "Could not detect preferred browser. Please set `${builtins.toString opt.uri.web-rad.browser}`.";
+      }
+      {
+        assertion =
+          1 >= length (
+            filter id [
+              cfg.uri.rad.browser.enable
+              cfg.uri.rad.vscode.enable
+            ]
+          );
+        message = "At most one `rad:`-URI handler may be enabled.";
+      }
+      {
+        assertion =
+          cfg.uri.rad.browser.enable -> hasSuffix publicExplorerSuffix cfg.settings.publicExplorer;
+        message = "${opt.uri.rad.browser.enable} is only compatible with a public explorer URL ending in '${publicExplorerSuffix}' but '${cfg.settings.publicExplorer}' does not end with '${publicExplorerSuffix}'.";
+      }
+    ];
+
+    home = {
+      packages = [ cfg.cli.package ];
+      file.".radicle/${configFile.name}".source = configFile.path;
+    };
+
+    programs.git.enable = mkDefault true;
```

Comment:
I know that having radicle without git is useless, but I am not sure whether it is okay to have this, the home-manager team must tell us here. @nix-community/home-manager-maintainers

Reply:
We have been preferring warnings/assertions > automatically enabling other modules.

Reply:
Also, that ping was to a large group (~250 people) instead of the smaller @nix-community/home-manager-committers if that is what you were trying to ping.